packages/runtime/src/interpreter/interpreter.ts

```diff
@@ -26,6 +37,85 @@ export class Interpreter {
     this.runtime = <Runtime>{};
   }
 
+  /**
+   * Queries ASA Definitions data by assetID
+   * @param assetId Asset Index
+   */
+  getAssetDef (assetId: number): AssetDef | undefined {
+    const accountAddr = this.runtime.ctx.state.assetDefs.get(assetId);
+    if (!accountAddr) return undefined;
+
+    let account = this.runtime.ctx.state.accounts.get(accountAddr);
+    account = this.runtime.assertAccountDefined(account);
+
+    return account.createdAssets.get(assetId);
+  }
+
+  /**
+   * Queries app (SSCAttributesM) from state. Throws TEAL.APP_NOT_FOUND if app is not found.
```

Comment:
Update the second part: explain when the app returns undefined and when it throws exception.